packages/http-server/test/integration/http-server.integration.ts

```diff
@@ -148,4 +158,30 @@ describe('HttpServer (integration)', () => {
       get(url, resolve).on('error', reject);
     });
   }
+
+  function givenHttpsServer(): HttpServer {
+    const keyPath = path.join(__dirname, 'key.pem');
+    const certPath = path.join(__dirname, 'cert.pem');
+    return new HttpServer(dummyRequestHandler, {
+      protocol: 'https',
+      key: keyPath,
+      cert: certPath,
+    });
+  }
+
+  function httpsGetAsync(host: string, port: number): Promise<IncomingMessage> {
+    const options = {
+      host: host,
+      port: port,
+      path: '/',
+      cert: fs.readFileSync(path.join(__dirname, 'cert.pem')),
+      agent: new Agent({
+        rejectUnauthorized: false,
```

Comment:
👍 

This is the solution I would like us to use instead of setting `NODE_TLS_REJECT_UNAUTHORIZED=0`.